tests/v2/validation/scaling/rke2_custom_cluster_add_nodes_test.go

```diff
@@ -0,0 +1,275 @@
+	client         *rancher.Client
+	clustersConfig *provisioninginput.Config
+}
+
+func (c *RKE2CustomClusterAddNodeTestSuit) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *RKE2CustomClusterAddNodeTestSuit) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+
+	c.clustersConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, c.clustersConfig)
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(c.T(), err)
+
+	c.client = client
+}
+
+func (c *RKE2CustomClusterAddNodeTestSuit) TestRKE2CustomClusterAddWorkerNode() {
+	subSession := c.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := c.client.WithSession(subSession)
+	require.NoError(c.T(), err)
+
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(c.T(), clusterName, "Cluster name is not provided")
+
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(c.T(), err)
+
+	daemonsetName := namegen.AppendRandomString("test-ds")
+
+	nodeProviderNames := c.clustersConfig.NodeProviders
+	for _, nodeProviderName := range nodeProviderNames {
+		externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviderName)
+		steveClient, err := client.Steve.ProxyDownstream(clusterID)
+		require.NoError(c.T(), err)
+
+		initialWorkerNodeCount, err := client.Management.Node.List(&types.ListOpts{
+			Filters: map[string]interface{}{
+				"clusterId": clusterID,
+				"worker":    true,
+			},
+		})
+		require.NoError(c.T(), err)
+
+		containerTemplate := workloads.NewContainer("rancher-test", "ranchertest/mytestcontainer", corev1.PullAlways, []corev1.VolumeMount{}, []corev1.EnvFromSource{})
+		podTemplate := workloads.NewPodTemplate([]corev1.Container{containerTemplate}, []corev1.Volume{}, []corev1.LocalObjectReference{}, nil)
+		daemonsetTemplate := workloads.NewDaemonSetTemplate(daemonsetName, workloadNS, podTemplate, true, nil)
+		daemonsetObj, err := steveClient.SteveType(workloads.DaemonsetSteveType).Create(daemonsetTemplate)
+		require.NoError(c.T(), err)
+		assert.Equal(c.T(), daemonsetObj.Name, daemonsetName)
+
+		err = charts.WatchAndWaitDaemonSets(client, clusterID, workloadNS, metav1.ListOptions{
+			FieldSelector: "metadata.name=" + daemonsetObj.Name,
+		})
+		require.NoError(c.T(), err)
+
+		c.validateDaemonset(client, clusterID, workloadNS, daemonsetObj)
+		logrus.Infof("daemonset created successfully")
+
+		newWorkerNodes, err := externalNodeProvider.NodeCreationFunc(client, []string{"--worker"}, []int32{1})
+		require.NoError(c.T(), err)
+
+		token, err := tokenregistration.GetRegistrationToken(client, clusterID)
+		require.NoError(c.T(), err)
+
+		for _, newWorkerNode := range newWorkerNodes {
+			command := fmt.Sprintf("%s %s --address %s", token.InsecureNodeCommand, "--worker", newWorkerNode.PublicIPAddress)
+			output, err := newWorkerNode.ExecuteCommand(command)
+			require.NoError(c.T(), err)
+			logrus.Info(output)
+		}
```

Comment:
just a note, possibly for future works; I feel like this would be highly re-usable and could be pulled out into a `ScaleRKE2K3SCustomCluster` public function.